odl/discr/lp_discr.py

```diff
@@ -1248,6 +1262,250 @@ def discr_sequence_space(shape, exponent=2.0, impl='numpy', **kwargs):
                          weighting='none', **kwargs)
 
 
+def uniform_discr_fromdiscr(discr, min_corner=None, max_corner=None,
+                            nsamples=None, cell_sides=None, exponent=2.0,
+                            interp='nearest', impl='numpy', **kwargs):
+    """Return a discretization based on an existing one.
+
+    The parameters that are explicitly given are used to create the
+    new discretization, and the missing information is taken from
+    the template space. See Notes for the exact functionality.
+
+    Parameters
+    ----------
+    discr : `DiscreteLp`
+        Uniformly discretized space used as a template.
+    min_corner : `float` or `tuple` of `float`
+        Minimum corner of the result.
+    max_corner : `float` or `tuple` of `float`
+        Maximum corner of the result.
+    nsamples : `int` or `tuple` of `int`
+        Number of samples per axis. For dimension >= 2, a tuple is
+        required.
+    exponent : positive `float`, optional
+        The parameter :math:`p` in :math:`L^p`. If the exponent is not
+        equal to the default 2.0, the space has no inner product.
+    interp : `str` or `sequence` of `str`, optional
+        Interpolation type to be used for discretization.
+        A sequence is interpreted as interpolation scheme per axis.
+
+            'nearest' : use nearest-neighbor interpolation
+
+            'linear' : use linear interpolation
+
+    impl : {'numpy', 'cuda'}, optional
+        Implementation of the data storage arrays
+    nodes_on_bdry : `bool` or `sequence`, optional
+        If a sequence is provided, it determines per axis whether to
```

Comment:
I feel like long docstrings like this should start with what it does, then go on with special cases. Here I have to read pretty much all of it before I learn if i should use this or not.